### PR TITLE
Add a file ring buffer writer and cleanup the logs

### PR DESCRIPTION
**Pull request by yuriykulikov**
Add a file ring buffer writer and cleanup the logs

### CODE REVIEW

**Comment by yuriykulikov** [comment]
@kolanse this PR fixes https://github.com/yuriykulikov/AlarmClock/issues/365

File: app/src/main/java/com/better/alarm/logger/LoggerModule.kt [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    object : LoggerFactory {
+      val logback: ILoggerFactory = configureLogback(get())
+      override fun createLogger(tag: String): Logger {
+        return Logger(logback.getLogger(tag))
+      }
+    }
+  }
+}
+
+private fun configureLogback(context: Context): ILoggerFactory {
```
Comment (yuriykulikov) [inline app/src/main/java/com/better/alarm/logger/LoggerModule.kt:30]: @kolanse this is the configuration for Logback, adding 2 appenders: LogCat and rolling file with 3 history items. `isCleanHistoryOnStart` doesn't slow the startup much so I chose to leave it.

Comment (kolanse) [reply app/src/main/java/com/better/alarm/logger/LoggerModule.kt:30]: Wow ... i really learnt a lot from the code and its implementation. Thank you very much for the tag!!!!

**Comment by kolanse** [comment]
Sorry for the late reply i have been super busy for a while

**Pull request merged**